mantidproject/mantid
# Figaro refactor loader

**Description of work.**
Adjusts `LoadILLReflectometry` to read new nexus files from figaro.



**To test:**
Use the 2 newly added test files.
1) Load both as direct beam:
```
ws1 = Load('709886')
ws2 = Load('709922')
```
Open the instrument viewer, with the help of pick tab make sure that the foreground line (the most prominent line) has `t=0`.

2) Load both as reflected beam, leaving to sample angle:
```
ws1 = Load('709886', Measurement='ReflectedBeam')
ws2 = Load('709922', Measurement='ReflectedBeam')
```
Now the foreground must have `t = 2 * Theta.wanted_theta` which you can find in the sample logs.

3) Load both as reflected beam, specifying BraggAngle:
```
ws1 = Load('709886', Measurement='ReflectedBeam', BraggAngle=%%%)
ws2 = Load('709922', Measurement='ReflectedBeam', BraggAngle=%%%)
```
Now the foreground must have `t = 2 * BraggAngle` that you passed.



Part of #30480. 


*This does not require release notes, since they will be added when the story is complete.*



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
